### DISCUSSION TITLE
Writing warn! and info! to different files

### DISCUSSION THREAD

**Discussion by jafioti** [tokio-rs/tracing#1750] [Q&A] [answered]
I am using the rolling appender to write to files, but I would like to write warn and info messages to different paths, from two different rolling appenders. Here's what I'm doing now:
```
let file_appender = tracing_appender::rolling::minutely("./logs/debug", "prefix.log");
    let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
    tracing_subscriber::fmt()
        .with_writer(non_blocking)
        .with_ansi(false)
        .with_max_level(Level::WARN)
        .init();
```

I tried to set a second appender as global, but you can only have one global writer at a time. Is there a way to split the output of info! to "/log/info" and warn to "/log/warn" ?

**Comment by hawkw** [comment, marked as answer]
Sending different kinds of events to different outputs can be accomplished with [`MakeWriter` filtering](https://docs.rs/tracing-subscriber/0.3.2/tracing_subscriber/fmt/writer/trait.MakeWriterExt.html). You can create a pair of rolling file appenders and combine them using the `MakeWriterExt` combinators.

In your case, to write warn and info messages to different rolling appenders, it might look something like this:
```rust
    use tracing_subscriber::fmt::writer::MakeWriterExt;

    let info_appender = tracing_appender::rolling::minutely("./logs/info", "prefix.log");
    let (info_appender, _info_guard) = tracing_appender::non_blocking(info_appender);
    let warn_appender = tracing_appender::rolling::minutely("./logs/info", "prefix.log");
    let (warn_appender, _warn_guard) = tracing_appender::non_blocking(warn_appender);

    let files = info_appender
        .with_min_level(Level::INFO)
        .or_else(warn_appender.with_max_level(Level::WARN));

    tracing_subscriber::fmt()
        .with_writer(files)
        .with_ansi(false)
        .with_max_level(Level::WARN)
        .init();
```

Also, note that the `with_max_level` filter you're adding to the subscriber itself will disable all `INFO` events...so the info file will be empty unless you remove that. :)

**Reply by jafioti** [reply]
Exactly what I wanted, thanks!

**Reply by hawkw** [reply]
Glad I could help!

We should probably also make it possible to combine the two files and then provide them both to a _single_ `non_blocking` writer, so that you don't have to spin up a separate worker thread for each file. But, the current API doesn't really permit that yet.
